src/jukebox/components/synchronisation/sync_shared/__init__.py

```diff
@@ -0,0 +1,355 @@
+cfg_sync_shared = jukebox.cfghandler.get_handler('sync_shared')
+cfg_cards = jukebox.cfghandler.get_handler('cards')
+
+
+class SyncShared:
+    """Control class for sync shared functionality"""
+
+    def __init__(self):
+        with cfg_main:
+            self._sync_enabled = cfg_main.setndefault('sync_shared', 'enable', value=False) is True
+        if self._sync_enabled:
+            logger.info("Sync shared activated")
+            with cfg_main:
+                config_file = cfg_main.setndefault('sync_shared', 'config_file',
+                                                    value='../../shared/settings/sync_shared.yaml')
+            try:
+                cfg_sync_shared.load(config_file)
+            except Exception as e:
+                logger.error(f"Error loading sync_shared config file. {e.__class__.__name__}: {e}")
+                return
+
+            with cfg_sync_shared:
+                self._sync_on_rfid_scan_enabled = (cfg_sync_shared.getn('sync_shared', 'on_rfid_scan_enabled', default=False)
+                                                is True)
+                if not self._sync_on_rfid_scan_enabled:
+                    logger.info("Sync on RFID scan deactivated")
+                self._sync_mode = cfg_sync_shared.getn('sync_shared', 'mode')
+                self._sync_remote_server = cfg_sync_shared.getn('sync_shared', self._sync_mode, 'server')
+                self._sync_remote_port = int(cfg_sync_shared.getn('sync_shared', self._sync_mode, 'port'))
+                self._sync_remote_timeout = int(cfg_sync_shared.getn('sync_shared', self._sync_mode, 'timeout'))
+                self._sync_remote_path = cfg_sync_shared.getn('sync_shared', self._sync_mode, 'path')
+
+                self._sync_is_mode_ssh = self._sync_mode == "ssh"
+                if self._sync_is_mode_ssh:
+                    self._sync_remote_ssh_user = cfg_sync_shared.getn('sync_shared', self._sync_mode, 'username')
+
+            components.rfid.reader.rfid_card_detect_callbacks.register(self.rfid_callback)
+        else:
+            logger.info("Sync shared deactivated")
+
+    def __exit__(self):
+        cfg_sync_shared.save(only_if_changed=True)
+
+    def rfid_callback(self, card_id: str, state: int):
+        if state == -1:
+            self.sync_card_database(card_id)
+
+    @plugs.tag
+    def sync_change_on_rfid_scan(self, option: str = 'toggle') -> None:
+        """
+        Change activation of 'on_rfid_scan_enabled'
+
+        :param option: Must be one of 'true', 'false', 'toggle'
+        """
+        if self._sync_enabled:
+
+            if option == 'true':
+                _new_state = True
+            elif option == 'false':
+                _new_state = False
+            elif option == 'toggle':
+                _new_state = not self._sync_on_rfid_scan_enabled
+            else:
+                logger.error(f"Invalid value '{option}' for 'option' in 'sync_change_on_rfid_scan'")
+                _new_state = None
```

Comment:
A variable that can have a boolean state should only be boolean.. what kind of state is `True` compared to a state `toggle`? It's also not easy to read from the code what `True` or `False` or even `Toggle` mean

Reply:
changed the options to "enable" and "disable" intead "true" and "false" for better readability